frontend/src/components/spatialChat/chatSettings/ChatSettings.tsx

```diff
@@ -0,0 +1,171 @@
+
+  const changeChatEditor = (editorType: ChatEditorType) => {
+    dispatch(changeEditorTypeAction(editorType));
+  };
+  const changeChatRadius = (radius: number) => {
+    dispatch(changeBroadcastRadius(radius));
+  };
+
+  const openSettings = useCallback(() => {
+    onOpen();
+  }, [onOpen]);
+
+  // Resetting the local states on close modal.
+  const closeSettings = useCallback(() => {
+    setLastEditorType(null);
+    setLastRadius(null);
+    onClose();
+  }, [onClose]);
+
+  const toast = useToast();
+
+  const validateSettings = () => {
+    if (lastRadius && (lastRadius < 80 || lastRadius > 1000)) {
+      toast({
+        title: 'Unable to update chat settings',
+        description: 'Broadcast Radius should be between 80 and 1000',
+        status: 'error',
+      });
+      return false;
+    }
+    return true;
+  };
+
+  const updateChatEditor = () => {
+    if (lastEditorType === '1') {
+      changeChatEditor(ChatEditorType.DEFAULT_EDITOR);
+    } else {
+      changeChatEditor(ChatEditorType.RICH_TEXT_EDITOR);
+    }
+  };
+  const updateChatRadius = () => {
+    if (lastRadius) {
+      changeChatRadius(lastRadius);
+    }
+  };
+
+  const updateSettings = () => {
+    try {
+      if (validateSettings()) {
+        updateChatEditor();
+        updateChatRadius();
+        toast({
+          title: 'Chat Settings Updated',
+          status: 'success',
+        });
+        closeSettings();
+      }
+    } catch (err) {
+      toast({
+        title: 'Unable to update chat settings',
+        description: err.toString(),
+        status: 'error',
+      });
+    }
+  };
+
+  return (
+    <>
+      <MenuItem onClick={openSettings}>
+        <Typography variant='body1'>Chat Settings</Typography>
+      </MenuItem>
+      <Modal isOpen={isOpen} onClose={closeSettings}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Edit Chat Settings</ModalHeader>
+          <ModalCloseButton />
+          <form
+            onSubmit={ev => {
+              ev.preventDefault();
+              updateSettings();
+            }}>
+            <ModalBody pb={8}>
+              <FormControl fullWidth>
+                <FormLabel>Chat Editor Type</FormLabel>
+                <Select
+                  value={lastEditorType || currentEditorType.toString()}
+                  onChange={e => setLastEditorType(e.target.value)}>
+                  {chateditorOptions.map((editor, index) => (
+                    <option value={editor.type.toString()} key={editor.type.toString()}>
+                      {editor.text}
+                    </option>
+                  ))}
+                </Select>
+              </FormControl>
+              <FormControl>
+                <FormLabel htmlFor='chatRadius'>Chat Radius</FormLabel>
+                <Input
+                  data-testid='chatRadius'
+                  id='chatRadius'
+                  placeholder='80'
+                  value={lastRadius || currentRadius}
```

Comment:
SUGGESTION: So it took some time for me to understand this line. From my understanding essentially, this is required because you never set the value of `lastEditorType` to `currentEditorType`. But this approach creates a problem. In you border-radius input, try deleting the entire entry. What happens is once you delete everything, your input defaults to 80 again (what if I want to put 120 in border radius. With the current way, you can't). This is because of the approach of doing `stateVal || reduxVal`.

So essentially, the user should be allowed to input any value (even if it's nothing) but we have validation in place that'll not submit them!

One approach that this is - 
```js
// In the component function
const currentRadius: number = useSelector(
    (state: RootState) => state.chat.settingChatBroadcastRadius,
  );
const [lastRadius, setLastRadius] = useState<string | null>();
useEffect(() => {
    setLastRadius(currentRadius.toString()) 
}, [currentRadius]);

// With this setup, the above line would essentially be 
<Input value={lastRadius} ...
```
With this, whenever the component renders, it always sets to whatever is pointed to by use-dispatch. The same logic applies to border-radius too. And if the border is set to empty, that's fine too. You validation should catch it.

Reply:
I have changed the logic now to this with some additional changes for the validations.